--- ./libs/locale/src/CMakeLists.txt.orig	2012-01-14 19:28:01.424416587 +0100
+++ ./libs/locale/src/CMakeLists.txt	2012-01-14 19:42:42.501651039 +0100
@@ -13,7 +13,11 @@
 else (ICU_FOUND AND ICU_I18N_FOUND)
   colormsg (CYAN "+-- ICU (unicode) not available for locale, disabling support.")
 endif (ICU_FOUND AND ICU_I18N_FOUND)
- 
+
+if (WIN32)
+  add_definitions (-DBOOST_LOCALE_NO_POSIX_BACKEND=1)
+  set (WIN32_SOURCES win32/collate.cpp win32/converter.cpp win32/lcid.cpp win32/numeric.cpp win32/win_backend.cpp std/codecvt.cpp std/collate.cpp std/converter.cpp std/numeric.cpp std/std_backend.cpp util/gregorian.cpp)
+endif (WIN32) 
 
 add_definitions (-DBOOST_THREAD_NO_LIB=1)
 boost_add_library (
@@ -27,6 +31,7 @@
   shared/localization_backend.cpp
   shared/message.cpp
   shared/mo_lambda.cpp
+  ${WIN32_SOURCES}
   util/codecvt_converter.cpp
   util/default_locale.cpp
   util/info.cpp
--- ./libs/timer/src/CMakeLists.txt.orig	2012-01-14 19:54:13.284819760 +0100
+++ ./libs/timer/src/CMakeLists.txt	2012-01-14 19:08:41.078415730 +0100
@@ -11,4 +11,5 @@
 
   SHARED_COMPILE_FLAGS "-DBOOST_TIMER_DYN_LINK=1"
   STATIC_COMPILE_FLAGS "-DBOOST_TIMER_STATIC_LINK=1"
+  DEPENDS boost_system boost_chrono
 )
--- ./libs/chrono/src/CMakeLists.txt.orig	2012-01-14 19:06:46.825514697 +0100
+++ ./libs/chrono/src/CMakeLists.txt	2012-01-14 19:07:00.369740065 +0100
@@ -12,4 +12,5 @@
 
   SHARED_COMPILE_FLAGS "-DBOOST_CHRONO_DYN_LINK=1"
   STATIC_COMPILE_FLAGS "-DBOOST_CHRONO_STATIC_LINK=1"
+  DEPENDS boost_system
 )
