--- tools/build/CMake/CMakeLists.txt.out_of_source	2012-01-27 19:36:23.900699615 +0100
+++ tools/build/CMake/CMakeLists.txt	2012-01-27 19:59:56.452813925 +0100
@@ -17,13 +17,13 @@
 # Provides version number to sphinx build
 #
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/docs/source/boost_cmake_version.py.in
-  ${CMAKE_CURRENT_SOURCE_DIR}/docs/source/boost_cmake_version.py)
+  ${CMAKE_CURRENT_BINARY_DIR}/docs/source/boost_cmake_version.py)
 
 #
 # Makefile for sphinx build with maintainer targets
 #
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/docs/Makefile.in
-  ${CMAKE_CURRENT_SOURCE_DIR}/docs/Makefile)
+  ${CMAKE_CURRENT_BINARY_DIR}/docs/Makefile)
 
 #
 # File used by BoostConfigVersion to import targets/libs/etc 
@@ -36,7 +36,7 @@
     BoostConfigVersionAgnostic)
   configure_file(
     ${CMAKE_CURRENT_SOURCE_DIR}/install_me/${file}.cmake.in 
-    ${CMAKE_CURRENT_SOURCE_DIR}/install_me/${file}.cmake @ONLY)
+    ${CMAKE_CURRENT_BINARY_DIR}/install_me/${file}.cmake @ONLY)
 endforeach()
 
 
@@ -44,8 +44,8 @@
 #  These are generated, version-specific
 #
 install(FILES
-  ${CMAKE_CURRENT_SOURCE_DIR}/install_me/BoostConfigVersion.cmake
-  ${CMAKE_CURRENT_SOURCE_DIR}/install_me/BoostConfig.cmake
+  ${CMAKE_CURRENT_BINARY_DIR}/install_me/BoostConfigVersion.cmake
+  ${CMAKE_CURRENT_BINARY_DIR}/install_me/BoostConfig.cmake
   DESTINATION ${BOOST_CMAKE_INFRASTRUCTURE_INSTALL_DIR}
   )
 
@@ -63,12 +63,12 @@
     set(agnostic_infra_dir ${BOOST_CMAKE_DRIVERS_INSTALL_DIR})
   endif()
   install(FILES
-    ${CMAKE_CURRENT_SOURCE_DIR}/install_me/BoostConfigAgnostic.cmake
+    ${CMAKE_CURRENT_BINARY_DIR}/install_me/BoostConfigAgnostic.cmake
     DESTINATION ${agnostic_infra_dir}
     RENAME BoostConfig.cmake
     )
   install(FILES
-    ${CMAKE_CURRENT_SOURCE_DIR}/install_me/BoostConfigVersionAgnostic.cmake
+    ${CMAKE_CURRENT_BINARY_DIR}/install_me/BoostConfigVersionAgnostic.cmake
     DESTINATION ${agnostic_infra_dir}
     RENAME BoostConfigVersion.cmake
     )
